### ISSUE TITLE
freebsd clog support 

### ISSUE THREAD

**Issue by simplexify** [ossec/ossec-hids#728]
We'd like to use OSSEC to monitor log files on a freebsd system (pfsense) that uses circular log (clog) files.

I've tried configuring clog support using the command option, but that doesn't seem to work.

Can anyone provide me with a pointer of how to use the command option for monitoring clog files (if at all possible)? Or alternatively, can clog support be added to OSSEC?

**Comment by ddpbsd** [comment]
(Untested) It looks like you could run `clog -f /path/to/file >> /path/to/another/file` to redirect the logs to a text file. You could possibly redirect the output to logger, but I bet that adds a bunch of junk you won't want (or will confuse OSSEC).

**Comment by simplexify** [comment]
great suggestion, works a treat as a workaround :+1:

**Comment by simplexify** [comment]
Doesn't quite fully work as expected in combination with logrotate to keep the size of that output file under control.

When the log rotates, clog -f doesn't start writing to the new file.
When I use pre- and postrotate to restart clog -f, it dumps the entire contents of the source clog file to the (new, rotated) output file, which makes the OSSEC agent resend all log entries to the OSSEC server, resulting in duplicate events.

What exactly do you have in mind with logger?

**Comment by ddpbsd** [comment]
```clog -f |logger```
But that will have the same problems, and more. Does sending the logs via syslog to OSSEC not work with pfsense?

**Comment by simplexify** [comment]
Do I understand correctly you are suggesting for pfsense to send logs via syslog to the OSSEC server (OSSIM in our case). This would not involve the OSSEC agent on the pfsense firewall, correct?

**Comment by ddpbsd** [comment]
That is correct. Skip the agent (or monitor no logs), and just use syslog. You miss out on syscheck and friends, but setting it up to monitor no logs should work just as well.
